SalesforceCommerceCloud/pwa-kit
# Multi-site implementation - dynamic config approach

**Prerequisite**
This PR assumed that the config file will be loaded dynamically as a result of work from #390. There is a short quip explaining the flow [here](https://salesforce.quip.com/ekejANjFSUO8)

## Context
According to 1 pager, we'd like to provide options 1a, and 2 for multisite feature
There are two options developers can choose to set up their multisite. 

- Option 1a: When a developer is standing up a multi-site project, they can configure a single codebase running on a single MRT environment/domain to work with multiple sites configured in Commerce Cloud business manager.
For example:

pwa-kit.mobifystorefront.com/us → RefArch site
pwa-kit.mobifystorefront.com/global → RefArchGlobal site
- Option 2: When a developer is standing up a multi-site project, they can configure a single codebase running on multiple MRT environments/domains to work with multiple sites configured in Commerce Cloud business manager
For example:

us.pwa-kit.mobifystorefront.com → RefArch site
intl.pwa-kit.mobifystorefront.com → RefArchGlobal site

# Description
This PR implemented the multi-site feature for The Retail React App

# Changes
   1. Add a new util function`resolveSiteFromUrl` that would help to determine from the sites list in the config file based on the current url. For example, if your current URL is `www.example.com/site-1/women/dresses`, this function will presume `site-1` is your site identifier, it can either be the `siteAlias` or the `siteId` that is being declared in the config. This function will use `site-1` and match the site in your config with that identifier and return the site object that has all information about the site to be used across the app.
   2. Add logic to `buildUrlWithUrlConfig` to reconstruct URLs that include site and locale. 
   3. Update `useNavigation` and Link component to use `buildUrlWithUrlConfig` for all navigation links/action
   4. Create a `useSite` hook that consumes `resolveSiteFromUrl` to be used for React components. One benefit of using this hook is it memoizes the site on the same location. 

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`



# How to Test-Drive This PR

- `npm start` in pwa
- Go to home page [localhost:3000](localhost:3000)
- You should see two buttons to navigate between two sites (RefArch and RefArchGlobal) as declared for the config inside `default.json`
- You should expect to see the site and locale identifiers in the urls when you navigate across the pages

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
